app/controllers/assessments_controller.rb

```diff
@@ -217,12 +224,33 @@ def update
       # Monitorees can't edit their own assessments, so the last person to touch this assessment was current_user
       assessment.who_reported = current_user.email
 
+      old_reported_at = assessment.reported_at
+
+      if params.permit(:reported_at)[:reported_at].present?
+        # Parse as a Time object first to capture timezone for history comment formatting
+        begin
+          rep_time = Time.strptime(params[:reported_at], '%Y-%m-%d %H:%M %z')
+          assessment.reported_at = rep_time.to_datetime.utc
+        rescue Date::Error => e
+          Rails.logger.info(
+            "AssessmentsController: Invalid date received for assessment.reported_at for patient ID: #{patient.id}. Error: #{e}"
+          )
+          return render json: { error: 'Assessment was unable to be saved due to invalid Report Date.' }, status: :bad_request
+        end
+      end
+      reported_at_changed = (old_reported_at - assessment.reported_at).abs >= 1
```

Comment:
Any reason to do math and a comparison instead of just comparison to see if they are not equal below?